packages/router/router.ts

```diff
@@ -1543,6 +1573,286 @@ export function createRouter(init: RouterInit): Router {
 }
 //#endregion
 
+////////////////////////////////////////////////////////////////////////////////
+//#region createStaticRouter
+////////////////////////////////////////////////////////////////////////////////
+
+export function createStaticHandler(init: StaticHandlerInit): StaticHandler {
+  invariant(
+    init.routes.length > 0,
+    "You must provide a non-empty routes array to createStaticRouter"
+  );
+
+  let dataRoutes = convertRoutesToDataRoutes(init.routes);
+
+  async function query(
+    request: Request
+  ): Promise<StaticHandlerState | Response> {
+    let { location, result } = await queryImpl(request);
+    if (result instanceof Response) {
+      return result;
+    }
+    // When returning StaticRouterState, we patch back in the location here
+    // since we need it for React Context.  But this helps keep our submit and
+    // loadRouteData operating on a Request instead of a Location
+    return { location, ...result };
+  }
+
+  async function queryRoute(request: Request, routeId: string): Promise<any> {
+    let { result } = await queryImpl(request, routeId);
+    if (result instanceof Response) {
+      return result;
+    }
+
+    // Pick off the right state value to return
+    let routeData = [result.errors, result.actionData, result.loaderData].find(
+      (v) => v
+    );
+    let value = Object.values(routeData || {})[0];
```

Comment:
Don't ever return `state` to a `queryRoute` call, and instead grab the raw data.  Should think about the non-Response error flow here.  When we pick off the "value" here we sort of lose visibility as to whether it was an error or not.  If user's are always throwing `Error` instances they could check themselves.  but what if they `throw 'not an error';`...?